app/controllers/admin_controller.rb

```diff
@@ -10,11 +10,22 @@ def patients
     @patient_count = Record.count
     @query_cache_count = HealthDataStandards::CQM::QueryCache.count
     @patient_cache_count = PatientCache.count
-    @provider_count = Provider.count
+    @provider_count = Provider.ne('cda_identifiers.root' => "Organization").count
+    @practice_count = Practice.count
+    @practices = Practice.asc(:name).map {|org| [org.name, org.id]}
   end
 
   def user_profile
     @user = User.find(params[:id])
+    @practice = @user.practice ? @user.practice.id : 'n/a'
+    @practices = Practice.asc(:name).map {|org| [org.name, org.id]}
+  end
+
+  def set_user_practice
+    @user = User.find(params[:user])
+    @user.practice = (params[:practice] != '')? Practice.find(params[:practice]) : nil
```

Comment:
giving the find() method a null or empty parameter throws an error. The usual ActiveRecord method find_by('') does not apply for Mongoid apparently..